src/coreclr/gc/gc.cpp

```diff
@@ -30280,7 +30283,11 @@ void gc_heap::mark_phase (int condemned_gen_number)
 #endif //FEATURE_EVENT_TRACE
 
         //decide on promotion
-        if (!settings.promotion)
+        int finalization_percent = (total_promoted_bytes > 0)
+            ? static_cast<int>((100 * finalization_promoted_bytes) / total_promoted_bytes) : 0;
+        dprintf (6666, ("finalization promotion: %d%% (%zd / %zd)", finalization_percent, finalization_promoted_bytes, total_promoted_bytes));
+	high_finalization_percentage = finalization_percent >= 90;
```

Comment:
90?  Microbenchmark is showing 95-98, so this works for it, but 90 is awfully high.

Reply:
6666 is probably not the right dprintf level here